dbcon/joblist/jlf_subquery.cpp

```diff
@@ -90,19 +90,32 @@ void getColumnValue(ConstantColumn** cc, uint64_t i, const Row& row, const strin
 
         case CalpontSystemCatalog::DECIMAL:
         case CalpontSystemCatalog::UDECIMAL:
-            data = row.getIntField(i);
-            oss << (data / IDB_pow[row.getScale(i)]);
-
-            if (row.getScale(i) > 0)
+            if (row.getColumnWidth(i) == datatypes::MAXDECIMALWIDTH)
             {
-                if (data > 0)
-                    oss << "." << (data % IDB_pow[row.getScale(i)]);
-                else if (data < 0)
-                    oss << "." << (-data % IDB_pow[row.getScale(i)]);
+                int128_t val;
+                row.getInt128Field(i, val);
+                IDB_Decimal dec(0, row.getScale(i), row.getPrecision(i), val);
+
+                oss << dec.toString(true);
+
+                *cc = new ConstantColumn(oss.str(), dec);
             }
+            else
+            {
+                data = row.getIntField(i);
+                oss << (data / IDB_pow[row.getScale(i)]);
 
-            *cc = new ConstantColumn(oss.str(),
-                                     IDB_Decimal(data, row.getScale(i), row.getPrecision(i)));
+                if (row.getScale(i) > 0)
+                {
+                    if (data > 0)
+                        oss << "." << (data % IDB_pow[row.getScale(i)]);
+                    else if (data < 0)
+                        oss << "." << (-data % IDB_pow[row.getScale(i)]);
+                }
+
+                *cc = new ConstantColumn(oss.str(),
+                                         IDB_Decimal(data, row.getScale(i), row.getPrecision(i)));
```

Comment:
Ohh, I didn't realize what you are doing here.
I've moved dataconvert::decimalToString into Decimal::toString() so it can print out both narrow and wide decimals and you need to only call Decimal ctor with appropriate arguments and call Decimal::toString() of the resulting instance calling ConstantColumn ctor.

Reply:
I agree. I am now constructing the IDB_Decimal object and calling the narrow decimal version of dec.toString().